### PR TITLE
[in_app_purchase] Added priceCurrencySymbol to SkuDetailsWrapper

### PR DESCRIPTION
**Pull request by renefloor**
This PR adds priceCurrencySymbol to the SkuDetailsWrapper. It uses the java function `Currency.getInstance(currencyCode).getSymbol()` to get the symbol.
We could change the platform_interface to also expose this in the ProductDetails.

Part of flutter/flutter#83984 (other part can be the platform_interface).

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

File: packages/in_app_purchase/in_app_purchase_android/pubspec.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: in_app_purchase_android
 description: An implementation for the Android platform of the Flutter `in_app_purchase` plugin. This uses the Android BillingClient APIs.
 repository: https://github.com/flutter/plugins/tree/master/packages/in_app_purchase/in_app_purchase_android
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+in_app_purchase%22
-version: 0.1.4+1
+version: 0.1.4+2
```
Comment (renefloor) [inline packages/in_app_purchase/in_app_purchase_android/pubspec.yaml:5]: If we first publish the new platform_interface we can upgrade `in_app_purchase_platform_interface` to 2.0.0 and add the property to the constructor of ProductDetails.

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
This PR seems to have a failing test that should be looked into (seems like the JAVA `Currency.getSymbol()` doesn't support the Euro symbol: https://docs.oracle.com/javase/7/docs/api/java/util/Currency.html#getSymbol() ) . Otherwise I think the changes look good.

```
io.flutter.plugins.inapppurchase.TranslatorTest > currencyCodeFromSymbol FAILED
    org.junit.ComparisonFailure: expected:<[€]> but was:<[EUR]>
        at org.junit.Assert.assertEquals(Assert.java:115)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at io.flutter.plugins.inapppurchase.TranslatorTest.currencyCodeFromSymbol(TranslatorTest.java:197)
```

**Comment by renefloor** [comment]
@mvanbeusekom Maybe it's better to first implement #4115 in this PR.

**Review by mvanbeusekom — APPROVED** [review]
LGTM

**Pull request merged**